ppafm/cli/generateElFF.py

```diff
@@ -1,178 +1,184 @@
-            geometry_format = "xsf"
-        elif(args.input.lower().endswith(".cube") ):
-            geometry_format = "cube"
-    else:
-        geometry_format = geometry_format.lower()
-    if(geometry_format=="xsf"):
-        print(">>> Loading Hartree potential from  ",args.input,"...")
-        print("Use loadXSF")
-        V, lvec, nDim, head = io.loadXSF(args.input)
-    elif(geometry_format=="cube"):
-        print(">>> Loading Hartree potential from ",args.input,"...")
-        print("Use loadCUBE")
-        V, lvec, nDim, head = io.loadCUBE(args.input)
-    else:
-        raise ValueError("""ERROR!!! Unknown or unsupported input ("-i") file format.\nSupported file fromats are:\n* cube\n* xsf""")
-
-    V *= -1 # Unit conversion, energy to potential (eV -> V)
-
-    if PPU.params['tip']==".py":
-        #import tip
-        exec(compile(open("tip.py", "rb").read(), "tip.py", 'exec'))
-        print(tipMultipole)
-        PPU.params['tip'] = tipMultipole
-        print(" PPU.params['tip'] ", PPU.params['tip'])
+    # Load parameters.
+    params_path = Path("params.ini") if Path("params.ini").is_file() else cpp_utils.PACKAGE_PATH / "defaults" / "params.ini"
+    common.loadParams(params_path)
+    common.apply_options(vars(args))
+
+    # Load species.
+    species_path = Path("atomtypes.ini") if Path("atomtypes.ini").is_file() else cpp_utils.PACKAGE_PATH / "defaults" / "atomtypes.ini"
+    common.loadSpecies(species_path)
+
+    subtract_core_densities = (args.doDensity) and (args.Rcore > 0.0) and (args.tip_dens is not None)
+    if subtract_core_densities:  # We do it here, in case it crash we don't want to wait for all the huge density files to load
+        if args.tip_dens is None:
+            raise Exception("Rcore>0 but no tip density provided!")
+        valence_electrons_dictionary = loadValenceElectronDict()
+        rs_tip, elems_tip = getAtomsWhichTouchPBCcell(args.tip_dens, Rcut=args.Rcore)
+
+    atoms_samp, _, lvec_samp = io.loadGeometry(args.input, format=args.input_format, params=common.params)
+    head_samp = io.primcoords2Xsf(atoms_samp[0], [atoms_samp[1], atoms_samp[2], atoms_samp[3]], lvec_samp)
+
+    # Load electrostatic potential.
+    loaders = {
+        "xsf": io.loadXSF,
+        "cube": io.loadCUBE,
+    }
+    input_format = args.input_format
+    try:
+        electrostatic_potential, lvec, n_dim, _ = loaders[input_format](args.input)
+    except KeyError:
+        input_format = args.input.split(".")[-1]
+        electrostatic_potential, lvec, n_dim, _ = loaders[input_format](args.input)
+
+    electrostatic_potential *= -1  # Unit conversion, energy to potential (eV -> V)
+
+    # To fix.
+    # if common.params["tip"] == ".py":
+    #     # import tip
+    #     exec(compile(open("tip.py", "rb").read(), "tip.py", "exec"))
+    #     print(tipMultipole)
+    #     common.params["tip"] = tipMultipole
+    #     print("params['tip'] ", common.params["tip"])
```

Comment:
What is this actually? Looks kinda hacky. Does removing this break any functionality? @ProkopHapala This is your code, I think.

Reply:
this allows the user to specify tip Multipole as dictionary in `tip.py` file

it is hacky, but why not

Reply:
cannot we just pass it as string like 
```
"{'dz2':-0.1,'pz':0.05,'s':-0.05}"
```
and than exec it? 

something like this:
```
if isinstance( common.params["tip"], str):
      exec( 'tipMultipole='+common.params["tip"] )
      common.params["tip"] = tipMultipole

```

Reply:
I made a new issue for it: #221.